homeassistant/components/shelly/light.py

```diff
@@ -369,3 +408,27 @@ def _update_callback(self) -> None:
         self.control_result = None
         self.mode_result = None
         super()._update_callback()
+
+
+class RpcShellyLight(ShellyRpcEntity, LightEntity):
+    """Entity that controls a light on RPC based Shelly devices."""
+
+    def __init__(self, wrapper: RpcDeviceWrapper, key: str) -> None:
+        """Initialize light."""
+        super().__init__(wrapper, key)
+        self._id = key[7:]
+
+    @property
+    def is_on(self) -> bool:
+        """If light is on."""
+        return bool(self.wrapper.device.status[self.key]["output"])
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on light."""
+        await self.set_state("Switch.Set", {"id": self._id, "on": True})
+        self.async_write_ha_state()
```

Comment:
Do you get a response before this RPC call is finished with the new state? And would that update also trigger a state update ?

Reply:
The response contains the previous switch state, but it is immediately followed by a state update, so I am removing the `self.async_write_ha_state()` and let the new state update to update the switch state.